sympy/calculus/euler.py

```diff
@@ -89,15 +89,17 @@ def euler_equations(L, funcs=(), vars=()):
         if not vars == f.args:
             raise ValueError("Variables %s don't match args: %s" % (vars, f))
 
-    order = max(len(d.variables) for d in L.atoms(Derivative)
-                if d.expr in funcs)
+    order = max([len(d.variables) for d in L.atoms(Derivative)
+                if d.expr in funcs] + [0])
 
     eqns = []
     for f in funcs:
         eq = diff(L, f)
         for i in range(1, order + 1):
```

Comment:
I think it would be better to compute the order for each `f` separately inside the loop.

How does this function behave if the "order" is different for different for each `f`?

Reply:
Did you mean to do something like the following?
```python
eqns = []
    for f in funcs:
        eq = diff(L, f)
        order = max([len(d.variables) for d in L.atoms(Derivative)
                        if d.expr == f] + [0])
        for i in range(1, order + 1):
            for p in combinations_with_replacement(vars, i):
                eq = eq + S.NegativeOne**i*diff(L, diff(f, *p), *p)
        new_eq = Eq(eq, 0)
        if isinstance(new_eq, Eq):
            eqns.append(new_eq)
```

Reply:
Yes, something like that. Are there any tests for examples where the order isn't the same for all functions?

Reply:
May be the following?
https://github.com/sympy/sympy/blob/9b16dae2ab364658ff75840bec7d4c2c8a6565b9/sympy/calculus/tests/test_euler.py#L48-L60

Reply:
It looks like the order is 1 for both x and y in that example.

Reply:
Does the following work?
```python
x, y, z = symbols('x y z')
f, g = Function('f', args=(x, y, z)), Function('g', args=(x, y, z))
f, g = f(), g()
expr2 = f.diff(x).diff(z)*g.diff(y)
print(euler_equations(expr2, (f, g)))
```

Reply:
I think so. Does that give different numbers for `order` for each function?

Reply:
> Does that give different numbers for `order` for each function?

Yes. 

```python
Function & Order:  f(x, y, z) 2
Function & Order:  g(x, y, z) 1
[Eq(Derivative(g(x, y, z), x, y, z), 0), Eq(-Derivative(f(x, y, z), x, y, z), 0)]
```

Reply:
I guess this case is like:
https://en.wikipedia.org/wiki/Euler%E2%80%93Lagrange_equation#Several_functions_of_several_variables_with_higher_derivatives

I haven't quite parsed the formulas there to convince myself that the output here is correct

Reply:
I think for now we can just keep the old behavior with 0 as default maximum order, so that we can get an empty list. Regarding the issue of functions with different orders, I can open an issue to discuss more on it.